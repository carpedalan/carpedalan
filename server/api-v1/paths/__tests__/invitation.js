import getSetup from '../../testUtils';

const mockSES = jest.fn(() => ({ promise: jest.fn() }));
const mockPromise = jest.fn(() => Promise.resolve({ farts: 'stinky' }));
jest.mock(
  'aws-sdk/clients/ses',
  () =>
    class SES {
      sendEmail(...args) {
        mockSES(...args);
        return this;
      }
      // eslint-disable-next-line
      promise() {
        return mockPromise();
      }
    },
);

const { afterAllCallback, validate, request } = getSetup({
  path: '/invitation/',
  method: 'post',
});

jest.mock('aws-cloudfront-sign', () => ({
  getSignedCookies: jest.fn(() => ({})),
}));

describe('POST /invitation', () => {
  afterAll(afterAllCallback);

  it('should require a name', async () => {
    const response = await request
      .post('/v1/invitation/')
      .set('Content-Type', 'application/json');
    validate(400, response);
    expect(response.body.errors[0].message).toBe(
      "should have required property 'name'",
    );
  });

  it('should require an email', async () => {
    const response = await request
      .post('/v1/invitation/')
      .set('Content-Type', 'application/json')
      .send({ name: 'farts' });
    validate(400, response);
    expect(response.body.errors[0].message).toBe(
      "should have required property 'email'",
    );
  });

  it('should have a valid email', async () => {
    const response = await request
      .post('/v1/invitation/')
      .set('Content-Type', 'application/json')
      .send({ name: 'farts', email: 'asdf' });
    validate(400, response);

    expect(response.body.errors[0].message).toBe('should match format "email"');
  });

  it('should send an email', async () => {
    await request
      .post('/v1/invitation/')
      .set('Content-Type', 'application/json')
      .send({ name: 'farts', email: 'leif@fo.co' });

    expect(mockSES).toHaveBeenCalledTimes(1);
    expect(mockSES.mock.calls[0]).toMatchSnapshot();
  });

  it.skip('should forward the error', async () => {
    mockPromise.mockImplementation = () => Promise.reject();
    const response = await request
      .post('/v1/invitation/')
      .set('Content-Type', 'application/json')
      .send({ name: 'farts', email: 'leif@fo.co' });
    expect(response.body).toBe('something');
  });
});
