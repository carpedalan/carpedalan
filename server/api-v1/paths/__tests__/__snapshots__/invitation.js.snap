// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /invitation should send an email 1`] = `
Array [
  Object {
    "Destination": Object {
      "ToAddresses": Array [
        "leifdalan@gmail.com",
      ],
    },
    "Message": Object {
      "Body": Object {
        "Html": Object {
          "Charset": "UTF-8",
          "Data": "undefined undefined with email: leif@fo.co requested access.",
        },
        "Text": Object {
          "Charset": "UTF-8",
          "Data": "Reqestsz",
        },
      },
      "Subject": Object {
        "Charset": "UTF-8",
        "Data": "Request to carpedalan.com",
      },
    },
    "ReplyToAddresses": Array [
      "no-reply@farts.com",
    ],
    "Source": "leifdalan@gmail.com",
  },
]
`;
